client/client.go

```diff
@@ -116,7 +117,14 @@ func CreateClient(kubeconfigPath string, overrides *clientcmd.ConfigOverrides) (
 
 				token, err := oauthConfig.TokenSource(context.TODO(), &oauth2.Token{RefreshToken: refreshToken}).Token()
 				if err != nil {
-					return nil, err
+					var e *oauth2.RetrieveError
+					if errors.As(err, &e) {
+						if strings.HasSuffix(err.Error(), `{"error":"invalid_grant"}`) {
```

Comment:
Is a raw JSON string check necessary?

Reply:
The response comes from the ODIC server in the form of an HTTP response body so some form for string check is necessary as we do not necessarily want to overwrite the other possible error responses. A proposed alternative would be `strings.Contains(err.Error(), "invalid_grant")`. Not too sure what the pros / cons are.

Reply:
After looking at `oauth2.RetrieveError` this is probably OKish, although it breaks easily.

Reply:
The response could be decoded (https://datatracker.ietf.org/doc/html/rfc6749#section-5.2), or at least changed to `strings.Contains(err.Error(), "invalid_grant")`. Current implementation is sensitive to whitespace in the JSON string.

Reply:
Agreed.